src/common/dnnl_thread.hpp

```diff
@@ -184,8 +184,12 @@ inline int dnnl_get_current_num_threads() {
 #define OMP_GET_NUM_THREADS() 1
 #endif
 
-// MSVC still supports omp 2.0 only
-#if defined(_MSC_VER) && !defined(__clang__) && !defined(__INTEL_COMPILER)
+// Disabling OMP SIMD feature on MSVC (still supports omp 2.0 only) and other
+// Windows debug builds completely due to incorrect results (_DEBUG is MSVC
+// specific).
+#if defined(_DEBUG) \
+        || (defined(_MSC_VER) && !defined(__clang__) \
+                && !defined(__INTEL_COMPILER))
```

Comment:
I'd go a safer way:
```suggestion
#if defined(_MSC_VER) && ((!defined(__clang__) && !defined(__INTEL_COMPILER)) || defined(_DEBUG))
```

And pairing undef must be matched as well.

Reply:
Done.